sdks/java/io/hcatalog/src/test/java/org/apache/beam/sdk/io/hcatalog/EmbeddedMetastoreService.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.hcatalog;
+
+import java.io.File;
+
+import org.apache.hadoop.hive.cli.CliSessionState;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.ql.CommandNeedRetryException;
+import org.apache.hadoop.hive.ql.Driver;
+import org.apache.hadoop.hive.ql.session.SessionState;
+import org.apache.hive.hcatalog.common.HCatUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of a light-weight embedded metastore.
+ * This class is a trimmed-down version of <a href="https://github.com/apache/hive/blob/master
+/hcatalog/core/src/test/java/org/apache/hive/hcatalog/mapreduce/HCatBaseTest.java">
+ * https://github.com/apache/hive/blob/master/hcatalog/core/src/test/java/org/apache/hive/hcatalog/mapreduce
+ * /HCatBaseTest.java </a>
+ */
+public class EmbeddedMetastoreService {
+
+  private static final Logger LOG = LoggerFactory.getLogger(EmbeddedMetastoreService.class);
+  private static final String BASE_DIR = System.getProperty("user.dir");
```

Comment:
I have refactored this class avoiding the use of any static and used JUnit tmp folders. I found a couple of odd things while using JUnit temporary folders - 
**_1._**  I had to explicitly invoke create() method as shown here - https://github.com/seshadri-cr/beam/blob/hiveio/sdks/java/io/hcatalog/src/test/java/org/apache/beam/sdk/io/hcatalog/HCatalogIOTest.java#L119. Otherwise when I access the folder instance to pass on the path as shown here - https://github.com/seshadri-cr/beam/blob/hiveio/sdks/java/io/hcatalog/src/test/java/org/apache/beam/sdk/io/hcatalog/HCatalogIOTest.java#L120 it fails with IllegalStateException - the temporary folder has not yet been created.
**_2_**  Had to explicitly specify this - https://github.com/seshadri-cr/beam/blob/hiveio/sdks/java/io/hcatalog/src/test/java/org/apache/beam/sdk/io/hcatalog/EmbeddedMetastoreService.java#L69 to create the workspace folders created under JUnit tmp folder. If the JUnit tmp folder is not empty at the end of the run, I noticed it does not delete it